cmd/cluster-agent/custommetrics/server.go

```diff
@@ -34,12 +34,13 @@ type DatadogMetricsAdapter struct {
 	basecmd.AdapterBase
 }
 
-// StartServer creates and start a k8s custom metrics API server
-func StartServer() error {
+// RunServer creates and start a k8s custom metrics API server
+func RunServer(ctx context.Context) error {
+	defer clearServerResources()
 	cmd = &DatadogMetricsAdapter{}
 	cmd.Flags()
 
-	provider, err := cmd.makeProviderOrDie()
+	provider, err := cmd.makeProviderOrDie(ctx)
```

Comment:
Small comment: it is strange to have a method call `makeProviderOrDie(ctx)` that returns an error and never die